substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ResourcesFeature.java

```diff
@@ -252,13 +253,16 @@ private static void scanDirectory(DebugContext debugContext, File root, Pattern[
         Map<String, List<String>> matchedDirectoryResources = new HashMap<>();
         Set<String> allEntries = new HashSet<>();
         ArrayList<File> queue = new ArrayList<>();
+        String separator = FileSystems.getDefault().getSeparator();
```

Comment:
There is a simpler way to write this without defining a new variable. `File.separator` is a system-dependent variable with the same value.